assets/js/extensions/google-analytics/index.ts

```diff
@@ -13,89 +17,217 @@ import {
+			items: [ getProductFieldObject( product, quantity ) ],
+		} );
+	}
+);
+addAction(
+	`${ actionPrefix }-set-cart-item-quantity`,
+	namespace,
+	( {
+		product,
+		quantity = 1,
+	}: {
+		product: CartResponseItem;
+		quantity: number;
+	} ): void => {
+		trackEvent( 'change_cart_quantity', {
+			event_category: 'ecommerce',
+			event_label: __(
+				'Change Cart Item Quantity',
+				'woo-gutenberg-products-block'
+			),
+			items: [ getProductFieldObject( product, quantity ) ],
+		} );
+	}
+);
+addAction(
+	`${ actionPrefix }-remove-cart-item`,
+	namespace,
+	( {
+		product,
+		quantity = 1,
+	}: {
+		product: CartResponseItem;
+		quantity: number;
+	} ): void => {
+		trackEvent( 'remove_from_cart', {
+			event_category: 'ecommerce',
+			event_label: __(
+				'Remove Cart Item',
+				'woo-gutenberg-products-block'
+			),
+			items: [ getProductFieldObject( product, quantity ) ],
+		} );
+	}
+);
 
-function initialize() {
-	// eslint-disable-next-line no-console
-	console.log( `Google Analytics Tracking initialized` );
-	addAction(
-		`${ actionPrefix }-list-products`,
-		namespace,
-		trackListProducts
-	);
-	addAction( `${ actionPrefix }-add-cart-item`, namespace, trackAddToCart );
-	addAction(
-		`${ actionPrefix }-set-cart-item-quantity`,
-		namespace,
-		trackChangeCartItemQuantity
-	);
-	addAction(
-		`${ actionPrefix }-remove-cart-item`,
-		namespace,
-		trackRemoveCartItem
-	);
-}
+/**
+ * Product Events.
+ */
+addAction(
+	`${ actionPrefix }-render-product-list`,
+	namespace,
+	( {
+		products,
+		listName = __( 'Product List', 'woo-gutenberg-products-block' ),
+	}: {
+		products: Array< ProductResponseItem >;
+		listName: string;
+	} ): void => {
+		if ( products.length === 0 ) {
+			return;
+		}
+		trackEvent( 'view_item_list', {
+			event_category: 'engagement',
+			event_label: __(
+				'Viewing products',
+				'woo-gutenberg-products-block'
+			),
+			items: products.map( ( product, index ) => ( {
+				...getProductImpressionObject( product, listName ),
+				list_position: index + 1,
+			} ) ),
+		} );
+	}
+);
+addAction(
+	`${ actionPrefix }-render-single-product`,
+	namespace,
+	( {
+		product,
+		listName,
+	}: {
+		product: ProductResponseItem;
+		listName: string;
+	} ): void => {
+		if ( product ) {
+			trackEvent( 'view_item', {
+				items: [ getProductImpressionObject( product, listName ) ],
+			} );
+		}
+	}
+);
+addAction(
+	`${ actionPrefix }-view-product`,
+	namespace,
+	( {
+		product,
+		listName,
+	}: {
+		product: ProductResponseItem;
+		listName: string;
+	} ): void => {
+		trackEvent( 'select_content', {
+			content_type: 'product',
+			items: [ getProductImpressionObject( product, listName ) ],
+		} );
+	}
+);
+addAction(
+	`${ actionPrefix }-search-for-product`,
+	namespace,
+	( { searchTerm }: { searchTerm: string } ): void => {
+		trackEvent( 'search', {
+			search_term: searchTerm,
+		} );
+	}
+);
 
-initialize();
+/**
+ * Exception events.
+ */
+addAction(
```

Comment:
I didn't find this documented in the tickets or the PR description. Let's add this at least in the PR description and mention a testing step for it

Reply:
I've added a mention for exceptions now. It was also in testing step 5. It's only used in checkout.